### PR TITLE
Modal support

### PR DESCRIPTION
**Pull request by VelvetThePanda**
https://github.com/discord/discord-api-docs/commit/9a86255edba4ce88e6a3f9f928fd14ec5d0097e0

- [x] Basic object support
- [x] Inbound interaction type
- [ ] API Tests

### CODE REVIEW

File: Backend/Remora.Discord.API/API/Objects/Interactions/Components/Component.cs [hunk 1]
```diff
@@ -42,5 +42,10 @@ public record Component
     Optional<IReadOnlyList<ISelectOption>> Options,
     Optional<string> Placeholder,
     Optional<int> MinValues,
-    Optional<int> MaxValues
+    Optional<int> MaxValues,
+    Optional<string> Title,
+    Optional<int> MinimumLength,
```
Comment (Nihlus) [inline Backend/Remora.Discord.API/API/Objects/Interactions/Components/Component.cs:47]: These should be "MinLength" and "MaxLength" to match the docs.

**Review by Nihlus — CHANGES_REQUESTED** [review]
Thanks! The PR isn't up to date with the public documentation provided by Discord, so you'll need to do a pass over this one. Please correct the compilation errors as well ;)

File: Backend/Remora.Discord.API.Abstractions/API/Objects/Interactions/Components/Modal/IModalComponent.cs [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+//  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+//
+
+using System.Collections.Generic;
+using Remora.Rest.Core;
+
+namespace Remora.Discord.API.Abstractions.Objects;
+
+/// <summary>
+/// Represents a modal component with one or more component fields.
```
Comment (Nihlus) [inline Backend/Remora.Discord.API.Abstractions/API/Objects/Interactions/Components/Modal/IModalComponent.cs:29]: This still references a modal as a component of some kind - please correct that.

**Review by Nihlus — CHANGES_REQUESTED** [review]

**Comment by VelvetThePanda** [comment]
@Nihlus round two of review 😄 | I'll have to investigate why those tests are failing (I did change something with the converter as per your request if memory serves, but I think I may have broken something there)

**Pull request merged**